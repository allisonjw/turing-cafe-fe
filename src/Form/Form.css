form {
    display: flex;
    align-items: center;
}

input {
    margin-left: 1%;
    height: 26px;
    width: 16%;
}

.form_btn {
    background-color: teal;
    height: 30px;
    width: 16%;
    color: white;
    margin-left: 1%;
}