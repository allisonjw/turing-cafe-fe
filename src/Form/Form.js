import React, {Component} from 'react';
import './Form.css';

class Form extends Component {
    constructor() {
        super()
        this.state = {
            name: '',
            date: '',
            time: '',
            number: ''
        }
    }

    handleChange = (e) => {
        this.setState({[e.target.name]: e.target.value})
    }

    submitReservation = (e) => {
        const newReservation = {
            id: Date.now(),
            name: this.state.name,
            date: this.state.date,
            time: this.state.time,
            number: this.state.number
        }
        this.props.addReservation(newReservation)
    }

    clearInputs = () => {
        this.setState({name: '', date: '', time: '', number: ''})
    }

    render() {
        return (
            <form>
                <input
                  className="form_input"
                  type="text"
                  name="name"
                  value={this.state.name}
                  onChange={e => this.handleChange(e)}
                ></input>
                <input
                  className="form_input"
                  type="text"
                  name="date"
                  value={this.state.date}
                  onChange={e => this.handleChange(e)}
                ></input>
                <input
                  className="form_input"
                  type="text"
                  name="time"
                  value={this.state.time}
                  onChange={e => this.handleChange(e)}
                ></input>
                <input
                  className="form_input"
                  type="text"
                  name="number"
                  value={this.state.number}
                  onChange={e => this.handleChange(e)}
                ></input>
                <button></button>
            </form>
        )
    }
}

export default Form;