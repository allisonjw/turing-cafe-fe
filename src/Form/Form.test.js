import React from 'react';
import {shallow} from 'enzyme';
import Form from './Form';

describe('Form', () => {
  let wrapper;

  beforeEach(() => {
    const mockSubmitReservation = jest.fn()
    wrapper = shallow(<Form submitReservation={mockSubmitReservation}/>)
  });

  it('should match all the data being passed', () => {
    expect(wrapper).toMatchSnapshot()
  });

  it('should update state on handleChange', () => {
      const mockEvent = {target: {name: 'name', value: 'Allison'}}
      const expected = 'Allison';

      wrapper.instance().handleChange(mockEvent);
      expect(wrapper.state('name')).toEqual(expected)
  })

});