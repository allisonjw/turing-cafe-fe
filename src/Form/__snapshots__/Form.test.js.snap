// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match all the data being passed 1`] = `
<form>
  <input
    className="form_input"
    name="name"
    onChange={[Function]}
    placeholder="Name"
    type="text"
    value=""
  />
  <input
    className="form_input"
    name="date"
    onChange={[Function]}
    placeholder="Date mm/dd"
    type="text"
    value=""
  />
  <input
    className="form_input"
    name="time"
    onChange={[Function]}
    placeholder="Time"
    type="text"
    value=""
  />
  <input
    className="form_input"
    name="number"
    onChange={[Function]}
    placeholder="Number"
    type="text"
    value=""
  />
  <button
    className="form_btn"
    onClick={[Function]}
  >
    Make Reservation
  </button>
</form>
`;
